src/pip/_internal/utils/misc.py

```diff
@@ -400,38 +400,21 @@ def encoding(self) -> str:  # type: ignore
         return self.orig_stream.encoding
 
 
-@contextlib.contextmanager
-def captured_output(stream_name: str) -> Generator[StreamWrapper, None, None]:
-    """Return a context manager used by captured_stdout/stdin/stderr
-    that temporarily replaces the sys stream *stream_name* with a StringIO.
-
-    Taken from Lib/support/__init__.py in the CPython repo.
-    """
-    orig_stdout = getattr(sys, stream_name)
-    setattr(sys, stream_name, StreamWrapper.from_stream(orig_stdout))
-    try:
-        yield getattr(sys, stream_name)
-    finally:
-        setattr(sys, stream_name, orig_stdout)
-
-
 def captured_stdout() -> ContextManager[StreamWrapper]:
```

Comment:
Do we use these often? If there’s only a couple of usages we can probably just inline the function.

Reply:
Ha no, they are not. I'll inline them then :+1: 

```
~/dev/oss/pip ❯ grep -R -P "captured_std(out|err)" pipdev/**.py -C1
pipdev/operations/install/wheel.py-from pip._internal.utils.filesystem import adjacent_tmp_file, replace
pipdev/operations/install/wheel.py:from pip._internal.utils.misc import captured_stdout, ensure_dir, hash_file, partition
pipdev/operations/install/wheel.py-from pip._internal.utils.unpacking import (
--
pipdev/operations/install/wheel.py-    if pycompile:
pipdev/operations/install/wheel.py:        with captured_stdout() as stdout:
pipdev/operations/install/wheel.py-            with warnings.catch_warnings():
--
pipdev/utils/misc.py-    "get_prog",
pipdev/utils/misc.py:    "captured_stdout",
pipdev/utils/misc.py-    "ensure_dir",
--
pipdev/utils/misc.py-
pipdev/utils/misc.py:def captured_stdout() -> ContextManager[StreamWrapper]:
pipdev/utils/misc.py-    """Capture the output of sys.stdout:
pipdev/utils/misc.py-
pipdev/utils/misc.py:    with captured_stdout() as stdout:
pipdev/utils/misc.py-        print('hello')
--
pipdev/utils/misc.py-
pipdev/utils/misc.py:def captured_stderr() -> ContextManager[StreamWrapper]:
pipdev/utils/misc.py-    """
pipdev/utils/misc.py:    See captured_stdout().
pipdev/utils/misc.py-    """
```

Reply:
See https://github.com/pypa/pip/pull/12639/commits/e9ad9797c59d92a4321f9b6bde506c14d112046d.